File: src/deltadak/HomeworkTask.java
package deltadak;

import java.io.Serializable;

//when transferred with the dragboard, the object is serialized
//which I think means that a new object is created and you lose the
// reference to the old one
//which I think should be fine here, as only content matters
class HomeworkTask implements Serializable {
    
    private boolean done;
    private String text;
    private String label;
    private String color;

    public HomeworkTask() {
        this.text = "default text";
        this.label = "default label";
        this.color = "white";
    }
    
    public HomeworkTask(final boolean done, final String text, final String label, final String color) {
        this.done = done;
        this.text = text;
        this.label = label;
        this.color = color;
    }
    
    public boolean getDone() {
        return done;
    }
    
    public void setDone(final boolean done) {
        this.done = done;
    }
    
    public String getText() {
        return text;
    }
    
    public void setText(final String text) {
        this.text = text;
    }
    
    public String getLabel() {
        return label;
    }
    
    public void setLabel(final String label) {
        this.label = label;
    }
    
    public String getColor() {
        return color;
    }
    
    public void setColor(final String color) {
        this.color = color;
    }
}

File: src/main/kotlin/nl/deltadak/plep/HomeworkTask.kt
package nl.deltadak.plep

import java.io.Serializable

/**
 * Represents a Task in the To Do list for a day.
 *
 * When transferred with the dragboard, the object is serialized which means that a new object is created and you lose the reference to the old one.
 * That is fine here, since only content matters. */
// todo remove "@JvmOverloads constructor" when no Java callers left
data class HomeworkTask @JvmOverloads constructor(
        /** Whether the task is finished or not. */
        var done: Boolean = false,
        /** Short description of the task. */
        var text: String = "",
        /** The label of a task, e.g. the course number. */
        var label: String = "",
        /** The id of the color of the task. */
        var colorID: Int = 4,
        /** Whether the task is expanded, showing subtasks. */
        var expanded: Boolean = false,
        /** id in the database of the task, which is there used to link subtasks to their parent. */
        var databaseID: Int = -1
        ) : Serializable
